### PR TITLE
Expose properties of ITrackingConsentFeature and cookie policy in Liquid

### LINKED ISSUE #17134: Expose ITrackingConsentFeature properties in Liquid
When building cookie-consent functionality with Liquid, it would be helpful to expose the functionality or at least some of the properties of the `ITrackingConsentFeature`. E.g. `CanTrack`, `IsConsentNeeded` and the consent cookie name and value would allow to build a JS consent dialog.

### PR DESCRIPTION
**Pull request by gvkries**
Fixes #17134

### CODE REVIEW

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/HttpRequestValue.cs [hunk 1]
```diff
@@ -12,9 +12,6 @@ internal sealed class HttpRequestValue : FluidValue
 {
     public override FluidValues Type => FluidValues.Object;
 
-    /// <summary>
-    /// Creates a new instance of a <see cref="HttpRequestValue"/> for the specified HTTP request.
-    /// </summary>
     public override bool Equals(FluidValue other)
```
Comment (gvkries) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/HttpRequestValue.cs:15]: Upps, this change is unrelated...

**Comment by hishamco** [comment]
IMHO this should be a part of Cookies Consent module

**Comment by MikeAlhayek** [comment]
@hishamco do we have Consent module? We seems to be using the consent interface is multiple places and there isn't a consent module for this.

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
+using Fluid;
+using Fluid.Values;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OrchardCore.Liquid;
+
+#nullable enable
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:10]: I don't think this is needed

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+namespace OrchardCore.DisplayManagement.Liquid.Values;
+
+/// <summary>
+/// Exposes <c>ITrackingConsentFeature</c> properties to Liquid.
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:15]:
```suggestion
/// Exposes <see href="ITrackingConsentFeature" />properties to Liquid.
```

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        return name switch
+        {
+            nameof(ITrackingConsentFeature.CanTrack) => BooleanValue.Create(feature.CanTrack),
+            nameof(ITrackingConsentFeature.HasConsent) => BooleanValue.Create(feature.HasConsent),
+            nameof(ITrackingConsentFeature.IsConsentNeeded) => BooleanValue.Create(feature.IsConsentNeeded),
+            "CookieName" => new StringValue(GetCookiePolicyOptions(context)?.ConsentCookie?.Name ?? string.Empty),
+            "CookieValue" => new StringValue(GetCookiePolicyOptions(context)?.ConsentCookieValue ?? string.Empty),
+
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:55]:
```suggestion
```

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+/// - HasConsent
+/// - IsConsentNeeded
+/// - CookieName
+/// - CookieValue
+/// </remarks>
+internal sealed class TrackingConsentValue : FluidValue
+{
+    public override FluidValues Type => FluidValues.Object;
+
+    public override bool Equals(FluidValue other) => other is TrackingConsentValue;
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:29]:
```suggestion
    public override bool Equals(FluidValue other) 
        => other is TrackingConsentValue;
```

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    }
+
+    private static ITrackingConsentFeature? GetTrackingFeature(TemplateContext context)
+    {
+        var ctx = context as LiquidTemplateContext
+            ?? throw new InvalidOperationException($"An implementation of '{nameof(LiquidTemplateContext)}' is required");
+
+        var httpContext = ctx.Services.GetRequiredService<IHttpContextAccessor>().HttpContext;
+
+        return httpContext?.Features.Get<ITrackingConsentFeature>();
```
Comment (MikeAlhayek) [inline src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:67]: This will never be null after GetRequiredService

```suggestion
        return httpContext.Features.Get<ITrackingConsentFeature>();
```

File: src/docs/reference/modules/Liquid/README.md [hunk 7]
```diff
@@ -559,6 +559,36 @@ Removes key from HttpContext.Items collection
 
 `{% httpcontext_remove_items "Item1" %}`
 
+### TrackingConsent
```
Comment (MikeAlhayek) [inline src/docs/reference/modules/Liquid/README.md:562]:
```suggestion
### Tracking Consent Cookie
```

**Review by MikeAlhayek — CHANGES_REQUESTED** [review]

**Comment by hishamco** [comment]
> @hishamco do we have Consent module? We seems to be using the consent interface is multiple places and there isn't a consent module for this.

Nope, I already did one, also Lombiq, that's why I'm asking here, unless we could create an abstraction project for it

**Comment by MikeAlhayek** [comment]
Then this PR is good where it is. Feel free to open another PR providing a consent module and move things to it if you add it

**Comment by hishamco** [comment]
My concern is anyone will add his Liquid filters or tags to be supported in his/her module, which doesn't make sense to me

**Comment by MikeAlhayek** [comment]
If there is already a module, that would be places in its corresponding module. If there isn't one, we can't request the PR creator to create a full blown module just to add Liquid filters.

If you create the consent module before releasing v3, we can move this filter there before the release.

**Comment by hishamco** [comment]
@gvkries it would be nice if you added a Cookie Consent module to make this functional

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs (hunk 6 above)
Comment (gvkries) [reply src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:67]: It can, if this is used without a request. It's the context at this point, not the service.

File: src/docs/reference/modules/Liquid/README.md (hunk 7 above)
Comment (gvkries) [reply src/docs/reference/modules/Liquid/README.md:562]: The header is the name of the liquid property, like e.g. `HttpContext` etc.

File: src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs (hunk 2 above)
Comment (gvkries) [reply src/OrchardCore/OrchardCore.DisplayManagement.Liquid/Values/TrackingConsentValue.cs:10]: I'm using some nullable annotations in this class.

**Review by MikeAlhayek — APPROVED** [review]

**Comment by MikeAlhayek** [comment]
Anything else @hishamco ?
